dataclients/kubernetes/routegroup.go

```diff
@@ -478,7 +480,34 @@ func splitHosts(hosts []string, domains []string) ([]string, []string) {
 	return internalHosts, externalHosts
 }
 
-func (r *routeGroups) convert(s *clusterState, df defaultFilters, loggingEnabled bool) ([]*eskip.Route, error) {
+// addRouteGroupTLS compares the RouteGroup host list and the RouteGroup TLS host list
+// and adds the TLS secret to the registry if a match is found.
+func (r *routeGroups) addRouteGroupTLS(ctx *routeGroupContext, tls *definitions.RouteTLSSpec) {
+	// Host in the tls section need to explicitly match the host in the RouteGroup
+	hostlist := compareStringList(tls.Hosts, ctx.routeGroup.Spec.UniqueHosts())
+	if len(hostlist) == 0 {
+		ctx.logger.Infof("No matching tls hosts found")
+		return
+	}
+
+	// Skip adding certs to registry since no certs defined
+	if tls.SecretName == "" {
+		ctx.logger.Infof("No tls secret defined for hosts - %s", tls.Hosts)
+		return
+	}
+
+	// Secrets should always reside in the same namespace as the RouteGroup
+	secretID := &definitions.ResourceID{Name: tls.SecretName, Namespace: ctx.routeGroup.Metadata.Namespace}
```

Comment:
I would use a plain value instead of a pointer here:
```
secretID := definitions.ResourceID{Name: tls.SecretName, Namespace: ctx.routeGroup.Metadata.Namespace}secret, ok := ctx.state.secrets[secretID]
```

The rest of the code won't change.